src/utils/positionpnl/positionpnl.js

```diff
@@ -0,0 +1,69 @@
+import {
+  retrieveData,
+  insertData,
+  putData
+} from '../network'
+import PositionPNL from '../../transactions/PostionPNL/PositionPNL'
+
+export function retrieve({ AMId }, callback) {
+  const params = {
+    AMaaSClass: 'positionpnl',
+    AMId,
+    bookId,
+    businessDate
+  }
+  let promise = retrieveData(params)
+    .then(result => {
+      result = result.map(positionpnl => _parsePositionPNL(pnl))
+      if (typeof callback === 'function') {
+        callback(null, result)
+      }
+      return result
+    })
+    if (typeof callback !== 'function') {
+      return promise
+    }
+}
+
+export function amend({ AMId, data }, callback) {
+  const params = {
+    AMaaSClass: 'positionpnl',
+    data: data,
+    AMId
+  }
+  let promise = putData(params).then(result => {
+    result = _parsePositionPNL(result)
+    if (typeof callback === 'function') {
+      callback(null, result)
+    }
+    return result
+  })
+  if (typeof callback !== 'function') {
+    return promise
+  }
+  promise.catch(error => callback(error))
+}
+
+export function insert({ AMId, data }, callback) {
+  const params = {
+    AMaaSClass: 'positionpnl',
+    data: data,
+    AMId,
+    upsert
```

Comment:
The `params` does not accept an `upsert` key - if you mean for the end result to look like

`https://.....com/123?upsert=true`,

you will need to use the `query` key.

Reply:
Sorry, I mean `queryParams`.